apps/docs/src/utils/useUser.jsx

```diff
@@ -0,0 +1,139 @@
+'use client'
+
+import { useEffect, useState, createContext, useContext, useMemo } from 'react';
+import { User } from '@supabase/supabase-auth-helpers/react';
+import { createPagesBrowserClient } from '@supabase/auth-helpers-nextjs';
+import * as Sentry from "@sentry/nextjs";
+
+/**
+ * @typedef {Object} UserContextType
+ * @property {boolean} isLoading
+ * @property {Session|null} session
+ * @property {User|null} user
+ * @property {Error|null} error
+ */
+export const UserContext = createContext(undefined);
+
+export const CustomUserContextProvider = (props) => {
+  const supabase = createPagesBrowserClient();
+  const [session, setSession] = useState(null);
+  const [user, setUser] = useState(null);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  let mounted;
+  
+  useEffect(() => {
+    mounted = true;
+    async function getSession() {
+      const {
+        data: { session },
+        error
+      } = await supabase.auth.getSession();
+
+      if (mounted) {
+        if (error) {
+          setError(error);
+          setIsLoading(false);
+          return;
+        }
+
+        setSession(session);
+        if (!session) setIsLoading(false); // if session is present, we set setLoading to false in the second useEffect
+      }
+    }
+    void getSession();
+    return () => {
+      mounted = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    const {
+      data: { subscription }
+    } = supabase.auth.onAuthStateChange((event, session) => {
+      if (session && (event === 'SIGNED_IN' || event === 'TOKEN_REFRESHED' || event === 'USER_UPDATED')) {
+        setSession(session);
+      }
+
+      if (event === 'SIGNED_OUT') {
+        setSession(null);
+      }
+    });
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+  
+  useEffect(() => {
+    async function getUserCustom() {
+      const { data: teams, teamsError } = await supabase
+        .from('users_teams')
+        .select('*')
+        .eq('user_id', session?.user.id) // Due to RLS, we could also safely just fetch all, but let's be explicit for sure
+      if (teamsError) Sentry.captureException(teamsError); 
+      
+      const { data: apiKeys, error: apiKeysError } = await supabase
```

Comment:
this is okay for now. There'll be issues when we introduce teams. I would add some comment about this

Reply:
I'm returning all teams and all api keys from provider, and displaying the first API key of the default team in the component, so it should be easy to adjust e..g to Dropdown in the future